CanaryValidator/Canary.Tests.ps1

```diff
@@ -94,47 +119,31 @@ while ($runCount -le $NumberOfIterations)
     {
         Remove-Item -Path $canaryUtilPath -Force -Recurse 
     }
-    New-Item -Path $canaryUtilPath -ItemType Directory
+    New-Item -Path $canaryUtilPath -ItemType Directory | Out-Null
 
     #
     # Start Canary 
     #
     $CanaryLogFile      = $CanaryLogPath + "\Canary-Basic$((Get-Date).ToString("-yyMMdd-hhmmss")).log"
+
+    Start-Scenario -Name 'Canary' -Type 'Basic' -LogFilename $CanaryLogFile -ContinueOnFailure $ContinueOnFailure
+
+    $SvcAdminEnvironmentName = $EnvironmentName + "-SVCAdmin"
+    $TntAdminEnvironmentName = $EnvironmentName + "-Tenant"
+
     if(-not $EnvironmentDomainFQDN)
     {
         $endptres = Invoke-RestMethod "${AdminArmEndpoint}/metadata/endpoints?api-version=1.0" -ErrorAction Stop 
         $EnvironmentDomainFQDN = $endptres.portalEndpoint
         $EnvironmentDomainFQDN = $EnvironmentDomainFQDN.Replace($EnvironmentDomainFQDN.Split(".")[0], "").TrimEnd("/").TrimStart(".") 
     }
 
-    Start-Scenario -Name 'Canary' -Type 'Basic' -LogFilename $CanaryLogFile -ContinueOnFailure $ContinueOnFailure
-
-    $SvcAdminEnvironmentName = $EnvironmentName + "-SVCAdmin"
-    $TntAdminEnvironmentName = $EnvironmentName + "-Tenant"
-
     Invoke-Usecase -Name 'CreateAdminAzureStackEnv' -Description "Create Azure Stack environment $SvcAdminEnvironmentName" -UsecaseBlock `
     {
```

Comment:
Instead of using just Add-AzureRmEnvironment we should leverage the Connect Tool so that it is tested as well.